### ISSUE TITLE
ui/ui_context_menu/contextmenu.js?_=1574886800200 404 (Not Found)

### ISSUE THREAD

**Issue by Hypnos3** [bartbutenaers/node-red-contrib-ui-contextmenu#19]
Context menu is not working.

I have import the example:
```
[{"id":"aa143ef1.58272","type":"ui_svg_graphics","z":"323ea0d9.cf8cc","group":"6daf1b39.51a344","order":1,"width":"14","height":"10","svgString":"<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:svg=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" preserveAspectRatio=\"none\" x=\"0\" y=\"0\" viewBox=\"0 0 900 710\" width=\"100%\" height=\"100%\">\n  <rect id=\"svgEditorBackground\" x=\"0\" y=\"0\" width=\"900\" height=\"710\" style=\"fill:none;stroke:none;\" />\n  <image width=\"889\" height=\"703\" id=\"background\" xlink:href=\"https://www.roomsketcher.com/wp-content/uploads/2016/10/1-Bedroom-Floor-Plans.jpg\" />\n  <circle id=\"mycircle\" cx=\"182.901\" cy=\"91.4841\" style=\"fill:rosybrown;stroke:black;stroke-width:1px;\" r=\"48\" /></svg>","clickableShapes":[{"targetId":"#mycircle","action":"click","payload":"camera_living","payloadType":"str","topic":"camera_living"}],"smilAnimations":[],"bindings":[{"selector":"#banner","bindSource":"payload.title","bindType":"text","attribute":""},{"selector":"#camera_living","bindSource":"payload.position.x","bindType":"attr","attribute":"x"},{"selector":"#camera_living","bindSource":"payload.camera.colour","bindType":"attr","attribute":"fill"}],"showCoordinates":false,"autoFormatAfterEdit":false,"outputField":"","editorUrl":"http://drawsvg.org/drawsvg.html","directory":"","name":"","x":4370,"y":450,"wires":[["e6045dac.cb243"]]},{"id":"f01245ca.882ee8","type":"ui_context_menu","z":"323ea0d9.cf8cc","group":"6daf1b39.51a344","order":5,"width":0,"height":-1,"fontSize":16,"position":"msg","outputField":"payload","xCoordinate":50,"yCoordinate":50,"menu":"msg","menuItems":[],"colors":"native","textColor":"#000000","backgroundColor":"#ffffff","borderColor":"#626262","intervalLength":"0","intervalUnit":"secs","name":"","x":4810,"y":450,"wires":[[]]},{"id":"e6045dac.cb243","type":"change","z":"323ea0d9.cf8cc","name":"","rules":[{"t":"set","p":"menu","pt":"msg","to":"[{\"text\":\"Options\",\"icon\":\"fa-list\",\"sub\":[{\"text\":\"Edit\",\"icon\":\"fa-edit\",\"topic\":\"edit\",\"payload\":[1,2,3,4,5],\"payloadType\":\"JSON\",\"outputField\":\"editArray\"},{\"text\":\"Cut\",\"icon\":\"fa-cut\",\"enabled\":true,\"topic\":\"cut\",\"payload\":\"true\",\"payloadType\":\"bool\"}]},{\"text\":\"---\"},{\"text\":\"Delete\",\"icon\":\"fa-trash\",\"enabled\":true,\"payload\":\"12\",\"payloadType\":\"num\"},{\"text\":\"---\"},{\"text\":\"Quit\",\"icon\":\"fa-times\",\"enabled\":false}]","tot":"json"}],"action":"","property":"","from":"","to":"","reg":false,"x":4590,"y":450,"wires":[["f01245ca.882ee8"]]},{"id":"6daf1b39.51a344","type":"ui_group","z":"","name":"Floorplan test","tab":"bf3a511d.fea31","disp":true,"width":"14","collapse":false},{"id":"bf3a511d.fea31","type":"ui_tab","z":"","name":"SVG","icon":"dashboard","disabled":false,"hidden":false}]
```
The floorplan is shown in the dashboard, but nothing happens on click the circle.

In the browser console I see that several files not be found:
![image](https://user-images.githubusercontent.com/12692680/69757869-59d5c000-115e-11ea-942d-f10d83e54e91.png)

```
VM217:1 GET http://192.168.178.205/addons/red/ui/ui_context_menu/contextmenu.js?_=1574886800200 404 (Not Found)
(anonymous) @ VM217:1
send @ app.min.js:20
ajax @ app.min.js:20
k.<computed> @ app.min.js:20
getScript @ app.min.js:20
eval @ VM319:44
map @ app.min.js:20
loadJavascriptAndCssFiles @ VM319:38
$scope.init @ VM319:137
fn @ VM320:2
$eval @ app.min.js:175
pre @ app.min.js:322
(anonymous) @ app.min.js:38
Ba @ app.min.js:110
p @ app.min.js:93
g @ app.min.js:85
(anonymous) @ app.min.js:85
(anonymous) @ app.min.js:600
$digest @ app.min.js:172
$apply @ app.min.js:176
(anonymous) @ app.min.js:597
(anonymous) @ app.min.js:598
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
Show 10 more frames
VM323:1 Uncaught SyntaxError: Unexpected token '<'
    at b (app.min.js:20)
    at Function.globalEval (app.min.js:20)
    at text script (app.min.js:20)
    at app.min.js:20
    at l (app.min.js:20)
    at XMLHttpRequest.<anonymous> (app.min.js:20)
b @ app.min.js:20
globalEval @ app.min.js:20
text script @ app.min.js:20
(anonymous) @ app.min.js:20
l @ app.min.js:20
(anonymous) @ app.min.js:20
load (async)
send @ app.min.js:20
ajax @ app.min.js:20
k.<computed> @ app.min.js:20
getScript @ app.min.js:20
eval @ VM319:44
map @ app.min.js:20
loadJavascriptAndCssFiles @ VM319:38
$scope.init @ VM319:137
fn @ VM320:2
$eval @ app.min.js:175
pre @ app.min.js:322
(anonymous) @ app.min.js:38
Ba @ app.min.js:110
p @ app.min.js:93
g @ app.min.js:85
(anonymous) @ app.min.js:85
(anonymous) @ app.min.js:600
$digest @ app.min.js:172
$apply @ app.min.js:176
(anonymous) @ app.min.js:597
(anonymous) @ app.min.js:598
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
Show 15 more frames
VM217:1 GET http://192.168.178.205/addons/red/ui/ui_context_menu/contextmenu.css 404 (Not Found)
```

An own try takes the same result.

I have no further idea what can I do.

**Comment by bartbutenaers** [comment]
Hi @Hypnos3 ,

Thanks for reporting this.  
Had some similar issues with other nodes.  Most of the time it happens for users that have customized the path in their Node-RED settings file:

```
ui: { path: "some_custom_path" },
```
Do you have something like that?

I have created a fix which you can install directly from Github:
```
npm install bartbutenaers/node-red-contrib-ui-contextmenu#UiPathFix
```
Hopefully that solves your problem.  Fingers crossed ...
Please let me know if it works!

Bart

**Comment by Hypnos3** [comment]
Yes, the Node-RED path is customized in my enviroment. It is an embeded device which runs on a raspberryPi on a Buildroot linux.

Your fix works perfect !

**Comment by bartbutenaers** [comment]
Thanks for the feedback!
The fix has published on NPM as version **1.0.5**
